File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string."""
        encoded_strs = ""

        for s in strs:
            encoded_strs += str(len(s)) + "|" + s

        return encoded_strs

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings."""
        decoded_strs = []
        i = 0

        while i < len(s):
            delim = s.find("|", i)
            length = int(s[i: delim])
            decoded_strs.append(s[delim + 1: delim + 1 + length])
            i = delim + 1 + length

        return decoded_strs


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.js
/**
 * Encodes a list of strings to a single string.
 *
 * @param {string[]} strs
 * @return {string}
 */
var encode = function (strs) {
    let encodedStrs = "";

    for (let str of strs) {
        encodedStrs += str.length + "#" + str;
    }

    return encodedStrs;
};

/**
 * Decodes a single string to a list of strings.
 *
 * @param {string} s
 * @return {string[]}
 */
var decode = function (s) {
    let decodedStrs = [];
    let i = 0;

    while (i < s.length) {
        const delimIndex = s.indexOf("#", i);
        const strLength = parseInt(s.substring(i, delimIndex));
        const decodedStr = s.substring(delimIndex + 1, delimIndex + 1 + strLength);
        decodedStrs.push(decodedStr);
        i = delimIndex + 1 + strLength;
    }

    return decodedStrs;
};

/**
 * Your functions will be called as such:
 * decode(encode(strs));
 */